src/test/regress/expected/failure_create_distributed_table_non_empty.out

```diff
@@ -0,0 +1,520 @@
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+-- kill as soon as the coordinator sends CREATE SCHEMA
+SELECT citus.mitmproxy('conn.onQuery(query="^CREATE SCHEMA").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+SELECT run_command_on_workers($$SELECT count(*) FROM information_schema.schemata WHERE schema_name = 'create_distributed_table_non_empty_failure'$$);
+ run_command_on_workers 
+------------------------
+ (localhost,57637,t,1)
+ (localhost,57640,t,0)
+(2 rows)
+
+-- cancel as soon as the coordinator sends CREATE SCHEMA
+SELECT citus.mitmproxy('conn.onQuery(query="^CREATE SCHEMA").cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+SELECT run_command_on_workers($$SELECT count(*) FROM information_schema.schemata WHERE schema_name = 'create_distributed_table_non_empty_failure'$$);
+ run_command_on_workers 
+------------------------
+ (localhost,57637,t,1)
+ (localhost,57640,t,1)
+(2 rows)
+
+SELECT run_command_on_workers($$DROP SCHEMA IF EXISTS create_distributed_table_non_empty_failure$$);
+      run_command_on_workers       
+-----------------------------------
+ (localhost,57637,t,"DROP SCHEMA")
+ (localhost,57640,t,"DROP SCHEMA")
+(2 rows)
+
+-- kill as soon as the coordinator sends begin
+SELECT citus.mitmproxy('conn.onQuery(query="^BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+ERROR:  connection error: localhost:57640
+DETAIL:  connection not open
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+SELECT run_command_on_workers($$SELECT count(*) FROM information_schema.schemata WHERE schema_name = 'create_distributed_table_non_empty_failure'$$);
```

Comment:
Maybe add a comment that this is expected, because the `SCHEMA` part went through but the transaction part failed. Is it a problem that we don't try to create the schema in the same transaction as we create the table in?

Reply:
Actually, strangely enough, we first create the schema, and then we again create the schema:

```
 (0,coordinator,"[initial message]")
 (0,worker,"['AuthenticationOk()', 'ParameterStatus(application_name=citus)', 'ParameterStatus(client_encoding=UTF8)', 'ParameterStatus(DateStyle=ISO, MDY)', 'ParameterStatus(integer_datetim
es=on)', 'ParameterStatus(IntervalStyle=postgres)', 'ParameterStatus(is_superuser=on)', 'ParameterStatus(server_encoding=UTF8)', 'ParameterStatus(server_version=XXX)', 'ParameterStatus(sessi
on_authorization=brian)', 'ParameterStatus(standard_conforming_strings=on)', 'ParameterStatus(TimeZone=XXX)', 'BackendKeyData(XXX)', 'ReadyForQuery(state=idle)']")
 **** HERE **** (0,coordinator,"['Query(query=CREATE SCHEMA IF NOT EXISTS test_test AUTHORIZATION brian)']")
 (0,worker,"['CommandComplete(command=CREATE SCHEMA)', 'ReadyForQuery(state=idle)']")
 (3,coordinator,"[""Query(query=BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, XX, 'XXXX-XX-XX XX:XX:XX.XXXXXX-XX');)""]")
 (3,worker,"['CommandComplete(command=BEGIN)', ""RowDescription(fieldcount=1,fields=['F(name=assign_distributed_transaction_id,tableoid=0,colattrnum=0,typoid=2278,typlen=4,typmod=-1,format_c
ode=0)'])"", 'DataRow(columncount=1,columns=[""C(length=0,value=b\\'\\')""])', 'CommandComplete(command=SELECT 1)', 'ReadyForQuery(state=in_transaction_block)']")
**** HERE ****  (3,coordinator,"[""Query(query=SELECT worker_apply_shard_ddl_command (100923, 'test_test', 'CREATE SCHEMA IF NOT EXISTS test_test AUTHORIZATION brian'))""]")
 (3,worker,"[""RowDescription(fieldcount=1,fields=['F(name=worker_apply_shard_ddl_command,tableoid=0,colattrnum=0,typoid=2278,typlen=4,typmod=-1,format_code=0)'])"", 'Backend(type=N,body=b\\
'SNOTICE\\\\x00VNOTICE\\\\x00C42P06\\\\x00Mschema ""test_test"" already exists, skipping\\\\x00Fschemacmds.c\\\\x00L122\\\\x00RCreateSchemaCommand\\\\x00\\\\x00\\')']")
 (3,worker,"['DataRow(columncount=1,columns=[""C(length=0,value=b\\'\\')""])', 'CommandComplete(command=SELECT 1)', 'ReadyForQuery(state=in_transaction_block)']")
 (3,coordinator,"[""Query(query=SELECT worker_apply_shard_ddl_command (100923, 'test_test', 'CREATE TABLE test_test.test_table (id integer)'))""]")
 (3,worker,"[""RowDescription(fieldcount=1,fields=['F(name=worker_apply_shard_ddl_command,tableoid=0,colattrnum=0,typoid=2278,typlen=4,typmod=-1,format_code=0)'])"", 'DataRow(columncount=1,c
olumns=[""C(length=0,value=b\\'\\')""])', 'CommandComplete(command=SELECT 1)', 'ReadyForQuery(state=in_transaction_block)']")
 (0,coordinator,"[""Query(query=BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, XX, 'XXXX-XX-XX XX:XX:XX.XXXXXX-XX');)""]")
 (0,worker,"['CommandComplete(command=BEGIN)', ""RowDescription(fieldcount=1,fields=['F(name=assign_distributed_transaction_id,tableoid=0,colattrnum=0,typoid=2278,typlen=4,typmod=-1,format_c
ode=0)'])"", 'DataRow(columncount=1,columns=[""C(length=0,value=b\\'\\')""])', 'CommandComplete(command=SELECT 1)', 'ReadyForQuery(state=in_transaction_block)']")
*** HERE *** (0,coordinator,"[""Query(query=SELECT worker_apply_shard_ddl_command (100925, 'test_test', 'CREATE SCHEMA IF NOT EXISTS test_test AUTHORIZATION brian'))""]")
 (0,worker,"[""RowDescription(fieldcount=1,fields=['F(name=worker_apply_shard_ddl_command,tableoid=0,colattrnum=0,typoid=2278,typlen=4,typmod=-1,format_code=0)'])"", 'Backend(type=N,body=b\\
'SNOTICE\\\\x00VNOTICE\\\\x00C42P06\\\\x00Mschema ""test_test"" already exists, skipping\\\\x00Fschemacmds.c\\\\x00L122\\\\x00RCreateSchemaCommand\\\\x00\\\\x00\\')']")
 (0,worker,"['DataRow(columncount=1,columns=[""C(length=0,value=b\\'\\')""])', 'CommandComplete(command=SELECT 1)', 'ReadyForQuery(state=in_transaction_block)']")
 (0,coordinator,"[""Query(query=SELECT worker_apply_shard_ddl_command (100925, 'test_test', 'CREATE TABLE test_test.test_table (id integer)'))""]")
 (0,worker,"[""RowDescription(fieldcount=1,fields=['F(name=worker_apply_shard_ddl_command,tableoid=0,colattrnum=0,typoid=2278,typlen=4,typmod=-1,format_code=0)'])"", 'DataRow(columncount=1,c
olumns=[""C(length=0,value=b\\'\\')""])', 'CommandComplete(command=SELECT 1)', 'ReadyForQuery(state=in_transaction_block)']")
 (0,coordinator,"['Query(query=COPY test_test.test_table_100925 (id) FROM STDIN WITH (FORMAT BINARY))']")
 (0,worker,"[""Backend(type=G,body=b'\\\\x01\\\\x00\\\\x01\\\\x00\\\\x01')""]")
 (3,coordinator,"['Query(query=COPY test_test.test_table_100923 (id) FROM STDIN WITH (FORMAT BINARY))']")
 (3,worker,"[""Backend(type=G,body=b'\\\\x01\\\\x00\\\\x01\\\\x00\\\\x01')""]")
 (0,coordinator,"[""CopyData(data=b'PGCOPY\\\\n\\\\xff\\\\r\\\\n\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00')"", ""CopyData(data=b'\\\\x00\\\\x01\\\\x00\\\\x00\\\\x00\\\\
x04\\\\x00\\\\x00\\\\x00\\\\x02')"", ""CopyData(data=b'\\\\xff\\\\xff')"", 'CopyDone()']")
 (0,worker,"['CommandComplete(command=COPY 1)', 'ReadyForQuery(state=in_transaction_block)']")
 (3,coordinator,"[""CopyData(data=b'PGCOPY\\\\n\\\\xff\\\\r\\\\n\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00\\\\x00')"", ""CopyData(data=b'\\\\x00\\\\x01\\\\x00\\\\x00\\\\x00\\\\
x04\\\\x00\\\\x00\\\\x00\\\\x03')"", ""CopyData(data=b'\\\\xff\\\\xff')"", 'CopyDone()']")
 (3,worker,"['CommandComplete(command=COPY 1)', 'ReadyForQuery(state=in_transaction_block)']")
 (3,coordinator,"[""Query(query=PREPARE TRANSACTION 'citus_0_7238_40_55')""]")
 (0,coordinator,"[""Query(query=PREPARE TRANSACTION 'citus_0_7238_40_57')""]")
 (0,worker,"['CommandComplete(command=PREPARE TRANSACTION)', 'ReadyForQuery(state=idle)']")
 (3,worker,"['CommandComplete(command=PREPARE TRANSACTION)', 'ReadyForQuery(state=idle)']")
 (3,coordinator,"[""Query(query=COMMIT PREPARED 'citus_0_7238_40_55')""]")
 (0,coordinator,"[""Query(query=COMMIT PREPARED 'citus_0_7238_40_57')""]")
 (3,worker,"['CommandComplete(command=COMMIT PREPARED)', 'ReadyForQuery(state=idle)']")
 (0,worker,"['CommandComplete(command=COMMIT PREPARED)', 'ReadyForQuery(state=idle)']")
 (0,coordinator,"['Terminate()']")
```

We do it three times, this is some strange behavior! Probably worth opening a ticket for?

Reply:
Ahh, that's I missed. Okay, I will check in detail an open a ticket for this.

Reply:
Yeah, we could open an issue. What I remembered related to that is we left it there because `GetTableDDLEvents()`, which is used in some other places as well, adds the `CREATE SCHEMA IF NOT EXISTS` for each shard creation. So, we should be careful while removing it, making sure that we don't break anyone who relies on that schema creation.

Reply:
opened the issue #2283